lib/hero_detail/hero_detail_component.dart

```diff
@@ -24,9 +24,10 @@ class HeroDetailComponent implements OnInit {
 
   @override
   Future<Null> ngOnInit() async {
-    var _id = _routeParams.get('id');
-    var id = int.parse(_id ?? '', onError: (_) => null);
```

Comment:
I thought it was not the point in this example code. And if we thought this case (value were either null or non-numeric value) seriously, the system should have raised an error.  But also I find the original toh-6 actually goes in the same way. Should I revert this part?